src/PostgREST/Error.hs

```diff
@@ -473,6 +500,7 @@ instance PgrstError Error where
   status (PgErr err)             = status err
   status PutMatchingPkError      = HTTP.status400
   status SingularityError{}      = HTTP.status406
+  status JSONParseError          = HTTP.status400
```

Comment:
```suggestion
  status JSONParseError          = HTTP.status500
```

This is similar to the `GucStatusError`, so this should be a Server error (500) instead of Client (400).

(BTW, I made a commit to the branch, so you should pull before making these changes).